docs/directory-structure.md

```diff
@@ -1,241 +1,205 @@
+
+This is a proposal for a new physical layout of the source code in this
+repo, meant to replace the somewhat ad-hoc layout of early 2021, in order
+to:
+
+  - Allow generalization to more languages than Plains Cree, and
+
+  - Address some other issues that have arisen with the source code layout.
+
+## The proposal
+
+### High-level decisions / assumptions
+
+  - In the file system, dictionary applications such as itwêwina are named
+    `sssttt`, where `sss` and `ttt` are three-letter [ISO 639-3 language
+    codes][639-3]. Example: `crkeng`.
+
+    The use of a technical abbreviation means that nobody will ever be
+    blocked from starting a dictionary for a new language pair on on
+    naming/branding questions.
+
+  - There will be at most one intelligent dictionary application for each
+    language pair. The code already supports multiple dictionary sources
+    within one application, such as both the *Cree: Words* and Maskwacîs
+    dictionaries for Plains Cree.
+
+  - Each site will be an independently deployed django project, and have an
+    independent database.
+
+    There are advantages and disadvantages to having a single django
+    process serving multiple sites, or having multiple sites share a single
+    database; however, this proposed compartmentalization should allow
+    reduced risk when experimenting with new languages.
+
+  - For now, all the code for all the dictionary applications will reside
+    in this git repo, and not be split into separate git repos.
+
+    That will make it much easier to move code around during development,
+    and to run tests across all dictionary applications when making
+    changes.
+
+    Some day, we hope, when mature and stable, morphodict could go on PyPI
+    as its own framework package with instructions on how to set up new
+    languages without being in this git repo. We are very far from that
+    point right now, but it’s something to keep in mind as a long-term
+    goal.
+
+[639-3]: https://iso639-3.sil.org/code_tables/639/data
+
+### Proposed layout
+
+    $repo
+    ├── .git/
+    ├── package.json                # dependencies for bundlers, JS/CSS frameworks
+    ├── Pipfile
+    ├── arpeng-manage.py            # manage.py scripts are at the top-level for easy access
+    ├── crkeng-manage.py
+    ├── crkfra-manage.py
+    ├── cwdeng-manage.py
+    ├── srseng-manage.py
+    └── src/
+        ├── CreeDictionary/         # existing code, eventually goes away
+        │   ├── __init__.py
+        │   ├── API/                # this name goes away :(
+        │   └── CreeDictionary/     # this goes away too
+        │       ├── __init__.py
+        │       ├── models.py
+        │       ├── views.py
+        │       └── …
+        │
+        ├── morphodict/             # python package for language-independent code
+        │   ├── __init__.py
+        │   ├── cvd/
+        │   ├── dictapp/            # django app with primary database tables
```

Comment:
@dwhieb What do you think about calling this [Django application ](https://docs.djangoproject.com/en/3.2/ref/applications/) `lexicon`?

This will affect the default name of database tables, e.g.,

```sql
SELECT text FROM lexicon_wordform;
```

or 

```sql
SELECT publisher FROM lexicon_dictionarysource;
```

---

It will also affect how the app is enabled:

```sql
# src/crkeng/site/settings.py
INSTALLED_APPS = [
   ...,
   "morphodict.lexicon.apps.LexiconAppConfig"
]
```

---

This will be visible in the Django Admin:

![Screenshot of Django Admin with  “LEXICONI” listed as the name of the primary application](https://user-images.githubusercontent.com/2294397/118305789-b2f5f880-b4a5-11eb-94a9-4a4354f2bc05.png)

Reply:
I think ‘lexicon’ is a much, *much* better than <sub><sup>‘dictapp’</sup></sub>. Thanks!

Reply:
I agree _lexicon_ is a big improvement!

If you want to get super precise, _dictionary_ could be used to refer to the Django app, and _lexicon_ to the actual database. You can think of a dictionary as a particular view on a lexicon—a lexicon can be presented in different ways. I'm totally fine with just using _lexicon_ though.